src/Form/Type/AdminType.php

```diff
@@ -110,7 +110,13 @@ static function (array $associationMapping): string {
                         $subject = $p->getValue($parentSubject, $parentPath.$path);
                     } catch (NoSuchIndexException $e) {
                         // no object here, we create a new one
-                        $subject = ObjectManipulator::setObject($admin->getNewInstance(), $parentSubject, $parentFieldDescription);
+                        $subject = $admin->getNewInstance();
+
+                        if (isset($options['by_reference']) && $options['by_reference']) {
```

Comment:
Not sure If it's better to do nothing if `by_reference` is not set ? Or adding a default value to true ?

Reply:
What could happen if we do nothing?

Reply:
If we do nothing it will be the same than some older SonataAdmin versions, parentObject won't be set to the children one during the form display so people can't use it to display some conditional fields.

1) No set: Behavior in the beginning of 2020
2) AddInstance: Behavior until the summer of 2020 => Allow conditionnal checks, but create #6166
3) SetObject: Behavior until now => Fix 6166, but create #6426

If I do nothing when the option is not set, it could lead to a regression for people with an older persistence bundle (that's why I added conflict) or custom persistence bundle.
But I agree that doing nothing is the more natural. Maybe should I add a NEXT MAJOR comment ? @jordisala1991

Reply:
What about:

1. we do not add conflict
2. we do here something to avoid changing default behavior for older persistence bundles (is there a difference here between older persisntence and new persistence releases)
3. maybe we add next major to clear this default thing? ( not sure about this one)

Reply:
I made more tests. The option is always set, so I removed the isset, but

Previously the persistence bundle didn't pass the option
Now: 
- If the option is set on the CollectionType, the persistence pass it
- If the option is not set, the persistence doesn't pass it BUT it used the default value (true) which is the same than the AdminType.

If someone use this code, it will lead to a behavior change if the option was false and not passed (because of an older persistence bundle version).
If I add something to say that the value will default to `false` if not passed, it won't fix the bug.

IMHO
- either we add conflict
- either we accept the minor change in the behavior (https://github.com/sonata-project/SonataAdminBundle/issues/6166 will appear again) and we recommend to update persistence bundle.

Reply:
Then lets leave the conflict. I tried to avoid it because it is hard to know how and when we can remove this conflict, probably on 4.x. But it could be better IMO to add that restriction here: https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/9e547e976f6d82bf2c40e33a1c637b9826cac60d/composer.json#L32

Reply:
I can't add the conflict on persistence bundle, since I want to block SontaAdmin 3.78+ with SonataDoctrineORM 3.23-

I leave the conflict. The PR is RTM.